lib/src/disposable/disposable.dart

```diff
@@ -214,19 +219,17 @@ class Disposable implements _Disposable, DisposableManagerV3 {
     }
     _isDisposing = true;
 
-    List<Future<dynamic>> futures = []
-      ..addAll(_internalDisposables.map((disposable) => disposable.dispose()))
-      ..add(onDispose());
-    _internalDisposables = [];
+    await Future.wait(_awaitableFutures);
+    _awaitableFutures.clear();
+
+    for (var disposable in _internalDisposables) {
+      await disposable.dispose();
+    }
```

Comment:
Yeah, that's what we thought. Once we get this squared away I'm going create a minimal test case and try to get to the bottom of it.